packages/carbon-web-components/src/components/tabs/tabs.ts

```diff
@@ -411,42 +411,64 @@ export default class CDSTabs extends HostListenerMixin(CDSContentSwitcher) {
     }
   }
 
-  render() {
+  /**
+   * Render the previous button if tablist is wider than container.
+   */
+  protected renderPreviousButton() {
     const {
       _isIntersectionLeftTrackerInContent: isIntersectionLeftTrackerInContent,
+    } = this;
+
+    return isIntersectionLeftTrackerInContent
+      ? html`
+          <button
+            part="prev-button"
+            tabindex="-1"
+            aria-hidden="true"
+            class="${prefix}--tab--overflow-nav-button ${prefix}--tabs__nav-caret-left ${prefix}--tab--overflow-nav-button--previous"
+            @click=${(_) =>
+            this._handleScrollButtonClick(_, {
+              direction: NAVIGATION_DIRECTION.Left,
+            })}>
+            ${ChevronLeft16()}
+          </button>
+        `
+      : null;
+  }
+
+  /**
+   * Render the next button if tablist is wider than container.
+   */
+  protected renderNextButton() {
+    const {
       _isIntersectionRightTrackerInContent: isIntersectionRightTrackerInContent,
+    } = this;
+
+    return isIntersectionRightTrackerInContent
+      ? html`
+          <button
+            part="next-button"
+            tabindex="-1"
+            aria-hidden="true"
+            class="${prefix}--tab--overflow-nav-button ${prefix}--tabs__nav-caret-right ${prefix}--tab--overflow-nav-button--next"
+            @click=${(_) =>
+              this._handleScrollButtonClick(_, {
+                direction: NAVIGATION_DIRECTION.Right,
+              })}>
+            ${ChevronRight16()}
+          </button>
+        `
+      : null;
+  }
+
+  render() {
+    const {
       _assistiveStatusText: assistiveStatusText,
       _handleSlotchange: handleSlotchange,
     } = this;
 
-    const previousButtonClasses = classMap({
-      [`${prefix}--tab--overflow-nav-button`]: true,
-      [`${prefix}--tabs__nav-caret-left`]: true,
-      [`${prefix}--tab--overflow-nav-button--previous`]: true,
-      [`${prefix}--tab--overflow-nav-button--hidden`]:
```

Comment:
This class doesn't get carried over to the `<button>` rendering in `renderPreviousButton`, is that intentional? I see that we render `null` if `isIntersectionLeftTrackerInContent`, so I'm guessing thats why we don't need it? If so, should be clean up styles that correspond to this class?

Reply:
Yep, no need to check for hidden if we don't render it at all. The corresponding styles actually come from the upstream `@carbon/styles` package: https://github.com/carbon-design-system/carbon/blob/main/packages/styles/scss/components/tabs/_tabs.scss#L104-L106